wgsl/index.bs

```diff
@@ -8247,20 +8247,36 @@ of declarations.
 
 ### Alias Analysis ### {#alias-analysis}
 
+#### Root Identifier #### {#root-ident-sec}
+
 [=Memory locations=] can be accessed during the execution of a function using [=memory views=].
-Within a function, each [=memory view=] has a particular <dfn noexport>root identifier</dfn>.
-The root identifier can be an [=originating variable=] or a [=formal parameter=]
-of [=pointer type=].
+Within a function, each [=memory view=] has a particular <dfn>root identifier</dfn>, which names
+the variable or formal parameter that first provides access to that memory in that function.
 
 Locally derived expressions of [=reference type|reference=] or
 [=pointer type|pointer=] type may introduce new names for a particular root identifier,
 but each expression has a statically determinable root identifier.
+
+<p algorithm="finding a root identifier">
+Given an expression |E| of [=pointer type|pointer=] or [=reference type=], the
+[=root identifier=] is the [=originating variable=] or [=formal parameter=] of [=pointer type=]
+found as follows:
+* If |E| is an identifier [=resolving=] to a variable, then the root identifier is that variable.
+* If |E| is an identifier [=resolving=] to a formal parameter of pointer type, then the root identifier is that formal parameter.
+* If |E| is an identifier [=resolving=] to a [=let-declaration=] with initializer |E2|, then the root identifier is the root identifier of |E2|.
+* If |E| is of the form `(`|E2|`)`, `&`|E2|, `*`|E2|, or |E2|`[`<var ignore>Ei</var>`]` then the root identifier is the root identifier of |E2|.
+* If |E| is a [[#vector-access-expr|vector access expression]] of the form |E2|.|swiz|, where |swiz| is a [=swizzle=] name, then the root identifer is the root identifier of |E2|.
+* If |E| is a [[#struct-access-expr|structure access expression]] of the form |E2|.<var ignore>member_name</var>, then the root identifer is the root identifier of |E2|.
+    <p>
+
+#### Aliasing #### {#aliasing}
+
 While the [=originating variable=] of a root identifier is a dynamic concept that
```

Comment:
This is the first instance in the section, it should be linked. I'm ok keeping or removing the next one, but keeping is probably more useful.

```suggestion
While the [=originating variable=] of a [=root identifier=] is a dynamic concept that
```